src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -52,7 +52,7 @@ internal NpgsqlBinaryExporter(NpgsqlConnector connector, string copyToCommand)
             _connector.Flush();
 
             CopyOutResponseMessage copyOutResponse;
-            var msg = _connector.ReadMessage();
+            var msg = _connector.ReadMessageWithoutCancellation();
```

Comment:
Didn't we say we want PG cancellation when reading messages when exporting?

Thinking about this again, we can say that we don't think PG cancellation makes sense with COPY, because we don't think PG will hang for any reason (after all it's just pushing out data), so we only hang because of network issues. If this is true (it makes sense to me), we can just do hard cancellation. Interestingly we still allow sending PG cancellation via the Cancel method, but that wouldn't be because of *hanging*, just because the user decided mid-way to stop the long process.

I'm still not completely sure here.

Reply:
But we're not reading here a CopyDataMessage. Is there a reason you think, that reading a CopyOutResponseMessage may hang?

Reply:
Also, just like `RawCopyStream` , here we're initializing the stream. Is it really OK to cancel, when we do not even know if the copy operation has really started?

Reply:
To be honest, I have no idea...

Let's go with your approach, and attempt cancellation only when reading CopyOutResponse. We can always adjust this later based on user feedback if we get any.